launcher/src/main/java/org/wso2/carbon/launcher/Main.java

```diff
@@ -156,6 +158,26 @@ private static void initAndVerifySysProps() {
             System.setProperty(PROFILE, DEFAULT_PROFILE);
         }
 
+        String runtimePath = System.getProperty(RUNTIME_PATH);
+        if (runtimePath == null || runtimePath.length() == 0) {
+            //If the runtime.home system property is not set, try to determine the runtime home using Main.class.
+            try {
+                Path tmpParent =
+                        Paths.get(Main.class.getProtectionDomain().getCodeSource().getLocation().toURI()).getParent();
+                if (tmpParent != null && (tmpParent = tmpParent.getParent()) != null &&
+                    (tmpParent = tmpParent.getParent()) != null) {
+                    runtimePath = tmpParent.toString();
+                } else {
+                    runtimePath = Paths.get(carbonHome).resolve("wso2").resolve(DEFAULT_PROFILE).toString();
+                }
+            } catch (URISyntaxException e) {
+                runtimePath = Paths.get(carbonHome).resolve("wso2").resolve(DEFAULT_PROFILE).toString();
+                logger.log(Level.WARNING, "Couldn't determine the runtime home. Runtime home is set to " + runtimePath,
+                           e);
+            }
+            System.setProperty(RUNTIME_PATH, runtimePath);
+        }
```

Comment:
We don't need this approach as RUNTIME_PATH will be set by the carbon.sh specific to that runtime.